core/src/backend/render.rs

```diff
@@ -213,25 +213,85 @@ pub enum JpegTagFormat {
 }
 
 /// Decoded bitmap data from an SWF tag.
-#[derive(Debug, Clone)]
+#[derive(Clone, Debug)]
 pub struct Bitmap {
-    pub width: u32,
-    pub height: u32,
-    pub data: BitmapFormat,
+    width: u32,
+    height: u32,
+    format: BitmapFormat,
+    data: Vec<u8>,
 }
 
-/// Decoded bitmap data from an SWF tag.
-/// The image data will have pre-multiplied alpha.
-#[derive(Debug, Clone)]
-pub enum BitmapFormat {
-    Rgb(Vec<u8>),
-    Rgba(Vec<u8>),
+impl Bitmap {
+    /// Ensures that `data` is the correct size for the given `width` and `height`.
+    pub fn from_data(width: u32, height: u32, format: BitmapFormat, mut data: Vec<u8>) -> Self {
+        let expected_len = match format {
```

Comment:
Minor suggestion: Some bits can be de-duplicated by introducing a `BitmapFormat::bytes_per_pixel` method that returns either 3 or 4, then use it here:
```rs
let expected_len = width as usize * height as usize * format.bytes_per_pixel();
```